### PR TITLE
Add support for raw block devices and enable block device tests on external k8s testing

### PR DESCRIPTION
**Pull request by davidz627**
Fixes: #64 

Add support for Raw Block devices.
Enable the external driver tests for block.

/assign @msau42

### CODE REVIEW

File: pkg/gce-pd-csi-driver/node.go [hunk 1]
```diff
@@ -95,18 +95,60 @@ func (ns *GCENodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePub
...
+
+		if err := ns.Mounter.Interface.MakeDir(targetPath); err != nil {
+			glog.Errorf("mkdir failed on disk %s (%v)", targetPath, err)
+			return nil, err
+		}
+	} else if blk := volumeCapability.GetBlock(); blk != nil {
+		glog.V(4).Infof("NodePublishVolume with block volume mode")
+
+		partition := ""
+		if part, ok := req.GetVolumeContext()[common.VolumeAttributePartition]; ok {
```
[the remark this answers is not in the document]
Comment (davidz627) [reply pkg/gce-pd-csi-driver/node.go]: The only broken part is specifying two partitions of the same disk in a pod: https://github.com/kubernetes/kubernetes/issues/57915

Our fs mount supports partitions as much as in-tree supports them (and will continue to have to for migration). If the in-tree block supports partitions we will have to support them for the driver as well

[the remark this answers is not in the document]
Comment (msau42) [reply pkg/gce-pd-csi-driver/node.go]: I'm unsure if it does. We should add test cases if we do...

[the remark this answers is not in the document]
Comment (davidz627) [reply pkg/gce-pd-csi-driver/node.go]: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/issues/287

**Comment by davidz627** [comment]
/hold
for https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/291

**Comment by davidz627** [comment]
@msau42 had to rebase PTAL

**Pull request merged**